content/blog/2018/08/2018-08-13-simple-pull-request-plugin-final-evaluation.adoc

```diff
@@ -0,0 +1,208 @@
+using git-plugin, PushCommand is used for this from git-plugin. credentialId,
+branch name and repository url for interacting with Github, Bitbucket, etc
+will be taken automatically from "Branch-Source" (Users have to fill this
+details of branch source in job configuration UI). (You can see
+link:https://github.com/Jenkinsci/simple-pull-request-job-plugin/blob/master/README.md[How to run the demo])
+. StepConfigurator: To generate pipeline code for all supported steps in Jenkins. This is using
+Jenkins configuration-as-code plugin (JCasC plugin) to configure a particular step object and
+then that step object is passed to `Snippetizer.object2Groovy()` method to generate the script of that step.
+
+=== Jenkinsfile.yaml example
+
+For the phase 1 prototype demonstration, the following yaml file was used.
+Note that this format is subject to change in the next phases of the project,
+as we formalise the yaml format definition.
+
+[source, yaml]
+----
+#  Docker image agent example
+agent:
+ label: my_label
+ customWorkspace: path_to_workspace
+ dockerImage: maven:3-alpine
+ args: -v /tmp:/tmp
+
+  tools:
+    maven : maven_3.0.1
+    jdk : jdk8
+
+configuration:
+  # Push PR changes to the target branch if the build succeeds.
+  # default value is false
+  pushPrOnSuccess: false
+
+  # Trusted user to approve pull requests
+  prApprovers:
+    - username1
+    - username2
+    - username3
+
+environment:
+  variables:
+    variable_1: value_1
+    variable_2: value_2
+
+  # Credentials contains only two fields. credentialId must be present in the Jenkins Credentials
+  credentials:
+    - credentialId : fileCredentialId
+      variable : FILE
+
+      # In user scripts Username and Password can be accessed by LOGIN_USR and LOGIN_PSW
+      # respectively as environment variales
+    - credentialId : dummyGitRepo
+      variable : LOGIN
+
+stages:
+  - name: stage1
+    agent: any
+    steps:
+      - sh: "scripts/hello"
+      - sleep:
+          time: 2
+          unit: SECONDS
+      - sleep: 2
+      - junit:
+          testResults: "target/**.xml"
+          allowEmptyResults: true
+          testDataPublishers:
+            - AutomateTestDataPublisher
+            - JunitResultPublisher:
+                urlOverride: "urlOverride"
+    # Post section for "stage1". All Conditions which are available in Jenkins
+    # declarative pipeline are supported
+    post:
+      failure:
+        - sh: "scripts/hello"
+
+# Outer post section. Just like declarative pipeline.
+post:
+  always:
+    - sh: "scripts/hello"
+----
+
+=== Tasks completed in Coding Phase 3
+
+
+. Add unit tests, JenkinsRule tests link:https://issues.jenkins-ci.org/browse/JENKINS-52495[JENKINS-52495]
+* [*] Refactor snippet generator to extensions (link:https://issues.jenkins-ci.org/browse/JENKINS-52491[JENKINS-52491])
+. Release 1.0 (link:https://issues.jenkins-ci.org/browse/JENKINS-52519[JENKINS-52519])
```

Comment:
the 1.0 release isn't likely to happen in this coding phase

also, you should reference JENKINS-52941 as future which _(which blocks the 1.0 release of this plugin)_

Reply:
releasing Beta version seems to be feasible in the current state